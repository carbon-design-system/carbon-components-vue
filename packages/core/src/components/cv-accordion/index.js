import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
