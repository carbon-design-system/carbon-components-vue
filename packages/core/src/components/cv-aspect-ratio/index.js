import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
