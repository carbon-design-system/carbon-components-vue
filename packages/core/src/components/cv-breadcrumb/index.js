import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
