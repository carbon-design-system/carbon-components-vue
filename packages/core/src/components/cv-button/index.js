import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
