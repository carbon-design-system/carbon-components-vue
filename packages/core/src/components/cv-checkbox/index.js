import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
