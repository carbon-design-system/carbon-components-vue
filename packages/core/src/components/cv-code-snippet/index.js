import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
