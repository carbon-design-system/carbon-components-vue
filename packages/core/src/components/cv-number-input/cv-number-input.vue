<template>
  <cv-wrapper :tag-type="formItem ? 'div' : ''" class="cv-number-input" :class="`${carbonPrefix}--form-item`">
    <div
      data-numberinput
      :class="[
        `${carbonPrefix}--number`,
        {
          [`${carbonPrefix}--number--light`]: theme === 'light',
          [`${carbonPrefix}--number--helpertext`]: isHelper,
          [`cv-number-input`]: !formItem,
          [`${carbonPrefix}--number--mobile`]: mobile,
        },
      ]"
      :data-invalid="isInvalid"
    >
      <label :for="uid" :class="`${carbonPrefix}--label`">{{ label }}</label>
      <div :class="`${carbonPrefix}--form__helper-text`" v-if="isHelper && mobile">
        <slot name="helper-text">{{ helperText }}</slot>
      </div>
      <div :class="`${carbonPrefix}--number__input-wrapper`">
        <button
          v-if="mobile"
          :class="`${carbonPrefix}--number__control-btn down-icon`"
          @click="doDown"
          type="button"
          :aria-label="ariaLabelForDownButton"
          :disabled="disabled"
        >
          <CaretDownGlyph />
        </button>
        <input
          :id="uid"
          type="number"
          :value="internalValue"
          v-bind="$attrs"
          v-on="inputListeners"
          :disabled="disabled"
          :step="step"
          :min="min"
          :max="max"
          ref="input"
        />
        <WarningFilled16 v-if="isInvalid && !mobile" :class="`${carbonPrefix}--number__invalid`" />
        <div :class="`${carbonPrefix}--number__controls`" v-if="!mobile">
          <button
            :class="`${carbonPrefix}--number__control-btn up-icon`"
            @click="doUp"
            type="button"
            :aria-label="ariaLabelForUpButton"
            :disabled="disabled"
          >
            <CaretUpGlyph />
          </button>
          <button
            :class="`${carbonPrefix}--number__control-btn down-icon`"
            @click="doDown"
            type="button"
            :aria-label="ariaLabelForDownButton"
            :disabled="disabled"
          >
            <CaretDownGlyph />
          </button>
        </div>
        <button
          v-else
          :class="`${carbonPrefix}--number__control-btn up-icon`"
          @click="doUp"
          type="button"
          :aria-label="ariaLabelForUpButton"
          :disabled="disabled"
        >
          <CaretUpGlyph />
        </button>
      </div>
      <div :class="`${carbonPrefix}--form-requirement`" v-if="isInvalid">
        <slot name="invalid-message">{{ invalidMessage }}</slot>
      </div>
      <div :class="`${carbonPrefix}--form__helper-text`" v-if="!isInvalid && isHelper && !mobile">
        <slot name="helper-text">{{ helperText }}</slot>
      </div>
    </div>
  </cv-wrapper>
</template>

<script>
import { uidMixin, themeMixin, carbonPrefixMixin, methodsMixin } from '../../mixins';
import CaretDownGlyph from '@carbon/icons-vue/es/caret--down/index';
import CaretUpGlyph from '@carbon/icons-vue/es/caret--up/index';
import WarningFilled16 from '@carbon/icons-vue/es/warning--filled/16';
import CvWrapper from '../cv-wrapper/_cv-wrapper';

export default {
  name: 'CvNumberInput',
  mixins: [uidMixin, themeMixin, carbonPrefixMixin, methodsMixin({ input: ['blur', 'focus'] })],
  components: { CaretDownGlyph, CaretUpGlyph, WarningFilled16, CvWrapper },
  inheritAttrs: false,
  props: {
    disabled: Boolean,
    formItem: { type: Boolean, default: true },
    helperText: { type: String, default: undefined },
    invalidMessage: { type: String, default: undefined },
    label: { type: String, default: '' },
    value: { type: [String, Number], default: '' },
    invalid: /* deprecate */ {
      type: Boolean,
      default: undefined,
      validator(val) {
        if (val !== undefined && process.env.NODE_ENV === 'development') {
          console.warn('CvNumberInput: invalid prop deprecated in favour of invalidMessage');
        }
        return true;
      },
    },
    ariaLabelForUpButton: { type: String, default: 'Increase number input' },
    ariaLabelForDownButton: { type: String, default: 'Decrease number input' },
    min: { type: [String, Number], default: undefined },
    max: { type: [String, Number], default: undefined },
    step: { type: [String, Number], default: undefined },
    mobile: Boolean,
  },
  data() {
    return {
      internalValue: 0,
      isHelper: false,
      isInvalid: false,
    };
  },
  mounted() {
    this.internalValue = this.valueAsString(this.value);
    this.checkSlots();
  },
  updated() {
    this.checkSlots();
  },
  watch: {
    value() {
      // NOTE: DELIBERATE USE OF != TO COMPARE this.interanlValue and this.value
      if (typeof this.value !== 'number' || this.internalValue != this.value) {
        // prevents this.value of 1 updating this.internalValue of 1.0
        // which improves the typing experience
        // does not matter if this.value is string or number
        this.internalValue = this.valueAsString(this.value);
      }
    },
  },
  computed: {
    // Bind listeners at the component level to the embedded input element and
    // add our own input listener to service the v-model. See:
    // https://vuejs.org/v2/guide/components-custom-events.html#Customizing-Component-v-model
    inputListeners() {
      return Object.assign({}, this.$listeners, {
        input: ev => this.onInput(ev.target.value),
      });
    },
  },
  methods: {
    onInput(val) {
      this.internalValue = val;
      this.emitValue();
    },
    checkSlots() {
      // NOTE: this.$slots is not reactive so needs to be managed on updated
      this.isInvalid = !!(this.$slots['invalid-message'] || (this.invalidMessage && this.invalidMessage.length));
      this.isHelper = !!(this.$slots['helper-text'] || (this.helperText && this.helperText.length));
    },
    doUp() {
      this.$refs.input.stepUp();
      this.onInput(this.$refs.input.value);
    },
    doDown() {
      this.$refs.input.stepDown();
      this.onInput(this.$refs.input.value);
    },
    emitValue() {
      if (typeof this.value === 'number') {
        if (this.internalValue != this.value) {
          const ePos = this.internalValue.indexOf('e-');
          const dotPos = this.internalValue.indexOf('.');
          if (ePos > -1 || dotPos > -1) {
            this.$emit('input', parseFloat(this.internalValue));
          } else {
            this.$emit('input', parseInt(this.internalValue));
          }
        }
      } else {
        this.$emit('input', this.internalValue);
      }
    },
    valueAsString(val) {
      let strVal;
      if (typeof val === 'number') {
        strVal = Number.isFinite(val) ? val.toString() : '';
      } else {
        strVal = val;
      }
      return strVal;
    },
  },
};
</script>
