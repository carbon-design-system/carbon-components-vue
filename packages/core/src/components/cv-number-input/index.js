import CvNumberInput from './cv-number-input';
import CvNumberInputSkeleton from './cv-number-input-skeleton';
export { CvNumberInput, CvNumberInputSkeleton };
