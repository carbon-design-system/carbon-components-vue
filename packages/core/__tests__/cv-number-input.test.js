// import { shallowMount as shallow } from '@vue/test-utils';
import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow, trigger, setProps } = awaitNextTick;
import { CvNumberInput, CvNumberInputSkeleton } from '@/components/cv-number-input';

describe('CvNumberInput', () => {
  // ***************
  // PROP CHECKS
  // ***************

  testComponent.propsHaveDefault(CvNumberInput, [
    'formItem',
    'ariaLabelForUpButton',
    'ariaLabelForDownButton',
    'label',
    'value',
  ]);
  testComponent.propsAreType(CvNumberInput, ['value'], [String, Number]);
  testComponent.propsAreType(CvNumberInput, ['formItem', 'invalid'], Boolean);
  testComponent.propsAreType(
    CvNumberInput,
    ['helperText', 'invalidMessage', 'label', 'ariaLabelForUpButton', 'ariaLabelForDownButton'],
    String
  );

  testComponent.propsHaveDefaultOfUndefined(CvNumberInput, ['helperText', 'invalidMessage']);

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should match snapshot when all props are default', async () => {
    const id = '1';
    const wrapper = await shallow(CvNumberInput, { propsData: { id } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot when it is not a formItem', async () => {
    const formItem = false;
    const id = '1';
    const wrapper = await shallow(CvNumberInput, { propsData: { formItem, id } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot when all props are provided', async () => {
    const formItem = false;
    const id = '1';
    const helperText = 'helperText';
    const invalidMessage = 'invalidMessage';
    const label = 'label';
    const value = 5;
    const ariaLabelForDownButton = 'ariaLabelForDownButton';
    const ariaLabelForUpButton = 'ariaLabelForUpButton';
    const wrapper = await shallow(CvNumberInput, {
      propsData: {
        formItem,
        id,
        helperText,
        invalidMessage,
        label,
        value,
        ariaLabelForDownButton,
        ariaLabelForUpButton,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot when value is String', async () => {
    const formItem = false;
    const id = '1';
    const value = '15';
    const wrapper = await shallow(CvNumberInput, { propsData: { formItem, id, value } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot when value is Number', async () => {
    const formItem = false;
    const id = '1';
    const value = 5;
    const wrapper = await shallow(CvNumberInput, { propsData: { formItem, id, value } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot with min, max and step as Strings', async () => {
    const formItem = false;
    const id = '1';
    const value = '15';
    const min = '-10';
    const max = '10';
    const step = '2';
    const wrapper = await shallow(CvNumberInput, { propsData: { formItem, id, value, min, max, step } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot with optional min, max and step as numbers', async () => {
    const formItem = false;
    const id = '1';
    const value = 15;
    const min = -10;
    const max = 10;
    const step = 2;
    const wrapper = await shallow(CvNumberInput, { propsData: { formItem, id, value, min, max, step } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot when light theme', async () => {
    const formItem = false;
    const id = '1';
    const theme = 'light';
    const wrapper = await shallow(CvNumberInput, { propsData: { formItem, id, theme } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot when invalid-message slot is provided', async () => {
    const id = '1';
    const value = 5;
    const propsData = { id, value };
    const wrapper = await shallow(CvNumberInput, {
      slots: {
        'invalid-message': '<div class="fake-slot">my invalid message</div>',
      },
      propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot when helper-text slot is provided', async () => {
    const id = '1';
    const value = 5;
    const propsData = { id, value };
    const wrapper = await shallow(CvNumberInput, {
      slots: {
        'helper-text': '<div class="fake-slot">my helper text</div>',
      },
      propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot when mobile variant requested', async () => {
    const id = '1';
    const value = 5;
    const mobile = true;
    const propsData = { id, value, mobile };
    const wrapper = await shallow(CvNumberInput, {
      slots: {
        'helper-text': '<div class="fake-slot">my helper text</div>',
      },
      propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************

  it('should emit input on doDown', async () => {
    const id = '1';
    const wrapper = await shallow(CvNumberInput, { propsData: { id } });
    await trigger(wrapper.find('.down-icon'), 'click');
    expect(wrapper.emitted().input).toBeTruthy();
  });

  it('should emit input on doUp', async () => {
    const id = '1';
    const wrapper = await shallow(CvNumberInput, { propsData: { id } });
    await trigger(wrapper.find('.up-icon'), 'click');
    expect(wrapper.emitted().input).toBeTruthy();
  });

  it('should emit input on change', async () => {
    const id = '1';
    const wrapper = await shallow(CvNumberInput, { propsData: { id } });
    await trigger(wrapper.find('input'), 'input');
    expect(wrapper.emitted().input).toBeTruthy();
  });

  it('should emit String when value prop is String', async () => {
    const id = '1';
    const value = '555';
    const wrapper = await shallow(CvNumberInput, { propsData: { id, value } });
    await trigger(wrapper.find('.up-icon'), 'click');
    expect(wrapper.emitted().input[0]).toEqual([(parseInt(value) + 1).toString()]);
  });

  it('should emit value increased by 1 on doUp', async () => {
    const id = '1';
    const value = 2;
    const wrapper = await shallow(CvNumberInput, { propsData: { id, value } });
    await trigger(wrapper.find('.up-icon'), 'click');
    expect(wrapper.emitted().input[0]).toEqual([value + 1]);
  });

  it('should emit value decreased by 1 on doDown', async () => {
    const id = '1';
    const value = 10;
    const wrapper = await shallow(CvNumberInput, { propsData: { id, value } });
    await trigger(wrapper.find('.down-icon'), 'click');
    expect(wrapper.emitted().input[0]).toEqual([value - 1]);
  });

  // the next four tests are commented out until JSDOM correctly
  // implements decimal floating point for <input> elements
  // see https://github.com/jsdom/jsdom/issues/2823

  // it('should emit value increased by 0.3 on doUp', async () => {
  //   const id = '1';
  //   const value = 0;
  //   const step = 0.3;
  //   const wrapper = await shallow(CvNumberInput, { propsData: { id, value, step } });
  //   await trigger(wrapper.find('.up-icon'), 'click');
  //   expect(wrapper.emitted().input[0]).toEqual([value + 0.3]);
  // });

  // it('should emit value decreased by 0.3 on doDown', async () => {
  //   const id = '1';
  //   const value = 1.2;
  //   const step = 0.3;
  //   const wrapper = await shallow(CvNumberInput, { propsData: { id, value, step } });
  //   await trigger(wrapper.find('.down-icon'), 'click');
  //   expect(wrapper.emitted().input[0]).toEqual([0.9]);
  // });

  // it('should emit value snapped to next higher value', async () => {
  //   const id = '1';
  //   const value = 0.5;
  //   const step = 0.3;
  //   const wrapper = await shallow(CvNumberInput, { propsData: { id, value, step } });
  //   await trigger(wrapper.find('.up-icon'), 'click');
  //   expect(wrapper.emitted().input[0]).toEqual([0.6]);
  // });

  // it('should emit value snapped to next lower value', async () => {
  //   const id = '1';
  //   const value = 0.5;
  //   const step = 0.3;
  //   const wrapper = await shallow(CvNumberInput, { propsData: { id, step, value } });
  //   await trigger(wrapper.find('.down-icon'), 'click');
  //   expect(wrapper.emitted().input[0]).toEqual([0.3]);
  // });

  it('should respect max value', async () => {
    const id = '1';
    const value = 9;
    const step = 2;
    const max = 10;
    const wrapper = await shallow(CvNumberInput, { propsData: { id, step, value, max } });
    await trigger(wrapper.find('.up-icon'), 'click');
    await trigger(wrapper.find('.up-icon'), 'click');
    expect(wrapper.emitted().input[0]).toEqual([10]);
    expect(wrapper.emitted().input[1]).toEqual([10]);
  });

  it('should respect min value', async () => {
    const id = '1';
    const value = 1;
    const step = 2;
    const min = 0;
    const wrapper = await shallow(CvNumberInput, { propsData: { id, step, value, min } });
    await trigger(wrapper.find('.down-icon'), 'click');
    await trigger(wrapper.find('.down-icon'), 'click');
    expect(wrapper.emitted().input[0]).toEqual([0]);
    expect(wrapper.emitted().input[1]).toEqual([0]);
  });

  it('should emit the correct value on input', async () => {
    const id = '1';
    const value = 2;
    const inputValue = 100;
    const wrapper = await shallow(CvNumberInput, { propsData: { id, value } });
    wrapper.find('input').setValue(inputValue);
    await trigger(wrapper.find('input'), 'input');
    expect(wrapper.emitted().input[0]).toEqual([inputValue]);
  });

  it('should watch value', async () => {
    const id = '1';
    const value1 = 2;
    const value2 = 80;
    const wrapper = await shallow(CvNumberInput, { propsData: { id, value: value1 } });
    expect(wrapper.vm.value).toEqual(value1);
    await setProps(wrapper, { value: value2 });
    expect(wrapper.vm.value).toEqual(value2);
  });
});

describe('CvNumberInputSkeleton', () => {
  // ***************
  // PROP CHECKS
  // ***************

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should match snapshot', async () => {
    const wrapper = await shallow(CvNumberInputSkeleton);
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // **********
});
