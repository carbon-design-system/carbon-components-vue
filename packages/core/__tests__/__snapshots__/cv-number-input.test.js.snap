// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvNumberInput should match snapshot when all props are default 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number">
      <!---->
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot when all props are provided 1`] = `
<cv-wrapper-stub tagtype="" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number bx--number--helpertext cv-number-input" data-invalid="true"><label for="1" class="bx--label">label</label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number" data-invalid="true">
      <warningfilled16-stub class="bx--number__invalid"></warningfilled16-stub>
      <div class="bx--number__controls"><button type="button" aria-label="ariaLabelForDownButton" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="ariaLabelForUpButton" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <div class="bx--form-requirement">invalidMessage</div>
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot when helper-text slot is provided 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number bx--number--helpertext"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number">
      <!---->
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <!---->
    <div class="bx--form__helper-text">
      <div class="fake-slot">my helper text</div>
    </div>
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot when invalid-message slot is provided 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number" data-invalid="true"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number" data-invalid="true">
      <warningfilled16-stub class="bx--number__invalid"></warningfilled16-stub>
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <div class="bx--form-requirement">
      <div class="fake-slot">my invalid message</div>
    </div>
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot when it is not a formItem 1`] = `
<cv-wrapper-stub tagtype="" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number cv-number-input"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number">
      <!---->
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot when light theme 1`] = `
<cv-wrapper-stub tagtype="" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number bx--number--light cv-number-input"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number">
      <!---->
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot when mobile variant requested 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number bx--number--helpertext bx--number--mobile"><label for="1" class="bx--label"></label>
    <div class="bx--form__helper-text">
      <div class="fake-slot">my helper text</div>
    </div>
    <div class="bx--number__input-wrapper"><button type="button" aria-label="Decrease number input" class="bx--number__control-btn down-icon">
        <subtract16-stub></subtract16-stub>
      </button> <input id="1" type="number">
      <!----> <button type="button" aria-label="Increase number input" class="bx--number__control-btn up-icon">
        <add16-stub></add16-stub>
      </button></div>
    <!---->
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot when value is Number 1`] = `
<cv-wrapper-stub tagtype="" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number cv-number-input"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number">
      <!---->
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot when value is String 1`] = `
<cv-wrapper-stub tagtype="" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number cv-number-input"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number">
      <!---->
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot with min, max and step as Strings 1`] = `
<cv-wrapper-stub tagtype="" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number cv-number-input"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number" step="2" min="-10" max="10">
      <!---->
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInput should match snapshot with optional min, max and step as numbers 1`] = `
<cv-wrapper-stub tagtype="" class="cv-number-input bx--form-item">
  <div data-numberinput="" class="bx--number cv-number-input"><label for="1" class="bx--label"></label>
    <!---->
    <div class="bx--number__input-wrapper">
      <!----> <input id="1" type="number" step="2" min="-10" max="10">
      <!---->
      <div class="bx--number__controls"><button type="button" aria-label="Decrease number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn down-icon">
          <subtract16-stub class="down-icon"></subtract16-stub>
        </button>
        <div class="bx--number__rule-divider"></div> <button type="button" aria-label="Increase number input" aria-live="polite" aria-atomic="true" class="bx--number__control-btn up-icon">
          <add16-stub class="up-icon"></add16-stub>
        </button>
        <div class="bx--number__rule-divider"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</cv-wrapper-stub>
`;

exports[`CvNumberInputSkeleton should match snapshot 1`] = `
<div class="cv-number-input bx--form-item"><label class="bx--label bx--skeleton"></label>
  <div class="bx--number bx--skeleton"></div>
</div>
`;
